#!/usr/bin/env python
PACKAGE = "pandora_slam_3d"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("inflation_size", int_t, 0, "inflation_size", 15,  3, 31)

edge_detection_method_enum = gen.enum([gen.const("CANNY", int_t, 0, "CANNY "),
  gen.const("SCHARR", int_t, 1, "SCHARR"),
  gen.const("CURVATURE", int_t, 2, "CURVATUREt")],
  "Edge detection method enum")
gen.add("edge_detection_method", int_t, 1, "edge_detection_method", 1, 0, 2, edit_method=edge_detection_method_enum)

gen.add("canny_low_threshold", int_t, 0, "canny_low_threshold", 10,  1, 100)
gen.add("canny_ratio", int_t, 0, "canny_ratio", 3,  1, 5)

gen.add("scharr_scale", int_t, 0, "scharr_scale", 1,  1, 3)
gen.add("scharr_delta", int_t, 0, "scharr_delta", 0,  1, 50)

gen.add("show_edges_image", bool_t, 0, "show_edges_image",  False)
gen.add("show_inflation_image", bool_t, 0, "show_inflation_image",  False)

exit(gen.generate(PACKAGE, "point_cloud_subsampler", "point_cloud_subsampler"))
